src/clj/db.clj

```diff
@@ -53,320 +131,308 @@
+          db-after (:db-after tx-result-question-set)]
+      (d/pull db-after [:question-set/id :question-set/name] [:question-set/id id])))
+
+
+  (add-user-answer!
+    [this user-id question-id answer]
+    (let [id (generate-id :answer/id)
+          tx-result (d/transact @(.conn this)
+                                [{:db/id -1
+                                  :answer/id id
+                                  :answer/question [:question/id question-id]
+                                  :answer/user [:user/id user-id]
+                                  :answer/answer answer}])
+          db-after (:db-after tx-result)]
+      (d/pull db-after [:answer/id {:answer/question [:question/id]}] [:answer/id id])))
+
+
+  (add-multiple-user-answers!
+    [this user-id answers]
+    (mapv
+      (fn [[question-id answer]]
+        (add-user-answer! this user-id question-id answer))
+      answers))
+
+
+  (get-all-answers
+    [this]
+    (d/q '[:find (pull ?e [:answer/id :answer/user :answer/question :answer/answer])
+           :where [?e :answer/id]]
+         @(.conn this)))
+
+
+  (get-answers-for-correction
+    [this answer-id]
+    (mapv first
+          (d/q '[:find (pull ?answer [:answer/id
+                                      :answer/points
+                                      :answer/answer
+                                      {:answer/question [:question/question-statement :question/points :question/evaluation-criteria]}])
+                 :in $ ?answer
+                 :where
+                 [?answer :answer/question ?question]
+                 [?question :question/type ?type]
+                 [(= ?type :question.type/free-text)]]
+               @(.conn this) [:answer/id answer-id])))
+
+
+  (get-corrections-of-answer
+    [this answer-id]
+    (map
+      #(zipmap [:correction/feedback :correction/timestamp] %)
+      (d/q '[:find ?corr-feedback ?timestamp
+             :in $ ?answer-id
+             :where
+             [?answer :answer/id ?answer-id]
+             [?correction :correction/answer ?answer ?tx]
+             [?tx :db/txInstant ?timestamp]
+             [?correction :correction/feedback ?corr-feedback]]
+           @(.conn this) answer-id)))
+
+
+  (add-correction!
+    [this ant-id {feedback :correction/feedback points :correction/points corr-id :corrector/id}]
+    (let [tx-result (d/transact @(.conn this)
+                                [{:db/id -1
+                                  :correction/feedback feedback
+                                  :correction/corrector [:user/id corr-id]
+                                  :correction/answer [:answer/id ant-id]}
+                                 {:answer/id ant-id
+                                  :answer/points points}])
+          db-after (:db-after tx-result)
+          ids (:tempids tx-result)]
+      (d/pull db-after [:correction/feedback {:correction/answer [:answer/points]}] (get ids -1)))))
+
+
+(def db (Database. conn))
```

Comment:
I'm not sure if that should stay here.
Isn't the core responsible for getting a database connection and explicitly handling the resulting instance to the layer of our onion, where it is needed?